packages/driver/src/cy/commands/cookies.ts

```diff
@@ -195,13 +195,16 @@ export default function (Commands, Cypress, cy, state, config) {
     getCookie (name, userOptions: Partial<Cypress.Loggable & Cypress.Timeoutable> = {}) {
```

Comment:
Changes to this file are adding retry logic to retry getting or setting cookies till you're on the correct origin.